backend/graphql/types/courseType.ts

```diff
@@ -1,13 +1,33 @@
 import { gql } from "apollo-server-express";
 
 const courseType = gql`
+  type ModuleDTO {
+    id: ID!
+    title: String!
+    description: String
+    image: String
+    previewImage: String
+    published: Boolean!
+    lessons: [String]
```

Comment:
If this is a return type, shouldn't they all be non-optional fields? Doesn't mean the user has to ask for them but they have to all exist

nitpick: for consistency, `ModuleResponseDTO`

Reply:
I made them optional because the pattern seemed to be that most fields were left optional, didn't really think through it, but yes making them non-optional makes more sense